zipkin-collector/rabbitmq/README.md

```diff
@@ -0,0 +1,57 @@
+Its only dependencies besides Zipkin core are the `slf4j-api` and the [RabbitMQ Java Client](https://github.com/rabbitmq/rabbitmq-java-client).
+
+### Configuration
+
+The following configuration can be set for the RabbitMQ Collector.
+
+Property | Environment Variable | Description
+--- | --- | ---
+`zipkin.collector.rabbitmq.addresses` | `RABBIT_ADDRESSES` | Comma-separated list of RabbitMQ addresses, ex. `localhost:5672,localhost:5673`
+`zipkin.collector.rabbitmq.concurrency` | `RABBIT_CONCURRENCY` | Number of concurrent consumers. Defaults to `1`
+`zipkin.collector.rabbitmq.connectionTimeout` | `RABBIT_CONNECTION_TIMEOUT` | Milliseconds to wait establishing a connection. Defaults to `60000` (1 minute)
+`zipkin.collector.rabbitmq.password` | `RABBIT_PASSWORD`| Password to use when connecting to RabbitMQ. Defaults to `guest`
+`zipkin.collector.rabbitmq.queue` | `RABBIT_QUEUE` | Queue from which to collect span messages. Defaults to `zipkin.spans`
+`zipkin.collector.rabbitmq.username` | `RABBIT_USER` | Username to use when connecting to RabbitMQ. Defaults to `guest`
+`zipkin.collector.rabbitmq.virtualHost` | `RABBIT_VIRTUAL_HOST` | RabbitMQ virtual host to use. Defaults to `/`
+
+### Caveats
+
+The configured queue will be idempotently declared as a durable queue.
+
+This collector uses one connection to RabbitMQ, with the configured `concurrency` number of threads
+each using one channel to consume messages.
+
+Consumption is done with `autoAck` on, so messages that fail to process successfully are not retried.
+
+## Encoding spans into RabbitMQ messages
+The message's body should be the bytes of an encoded list of spans.
+
+### JSON
+A list of Spans in JSON. The first character must be '[' (decimal 91).
+
+`SpanBytesEncoder.JSON_V2.encodeList(spans)` performs the correct JSON encoding.
+
+## Local testing
+
+The following assumes you are running an instance of RabbitMQ locally on the default port (5672).
+You can download and install RabbitMQ following [instructions available here](https://www.rabbitmq.com/download.html).
+With the [RabbitMQ Management CLI](https://www.rabbitmq.com/management-cli.html) you can easily publish
+one-off spans to RabbitMQ to be collected by this collector.
+
+1. Start RabbitMQ server
+2. Start Zipkin server
+```bash
+$ RABBIT_ADDRESSES=localhost java -jar zipkin.jar
+```
+3. Save an array of spans to a file like `sample-spans.json`
+```json
+[{"traceId":"73321ecb02595c39","id":"73321ecb02595c39","name":"get","annotations":[{"timestamp":1505990621526000,"value":"sr","endpoint":{"serviceName":"service","ipv4":"127.0.0.1","port":8080}},{"timestamp":1505990621527000,"value":"ss","endpoint":{"serviceName":"service","ipv4":"127.0.0.1","port":8080}}],"binaryAnnotations":[{"key":"http.method","value":"GET","endpoint":{"serviceName":"service","ipv4":"127.0.0.1","port":8080}}]}]
+```
+4. Publish them using the CLI 
+```bash
+$ rabbitmqadmin publish exchange=amq.default routing_key=zipkin.spans < sample-spans.json
```

Comment:
I think the topic-like things have always defaulted to "zipkin" If that becomes ambiguous, we can append for whatever else we do. I'd prefer consistency and something obvious to guess.

(you might suggest this consistency is violated in ES, but that's a bit different as we have no dependency link transport, even if we have dependency link storage. if we did, we'd have to append to all the others, too.